### ISSUE TITLE
Jazzy can't find RealmCore.modulemap in DerivedData

### ISSUE THREAD

**Issue by Mozahler** [realm/jazzy#1225]
Here's a variation on an earlier issue I opened (#1220).
It was closed as answered, **but never answered for me**. I still can't use Jazzy. This is the first command that is issued, and it produces an error:

CompileC /Users/Alex/Library/Developer/Xcode/DerivedData/Pent-aedkuecfpwytirevchqirntnielu/SourcePackages/checkouts/realm-cocoa/build/Realm.build/Release-iphoneos/Realm.build/Objects-norma    l/arm64/collection_notifications.o /Users/Alex/Library/Developer/Xcode/DerivedData/Pent-aedkuecfpwytirevchqirntnielu/SourcePackages/checkouts/realm-cocoa/Realm/ObjectStore/src/collection_n    otifications.cpp normal arm64 c++ com.apple.compilers.llvm.clang.1_0.compiler (in target 'Realm' from project 'Realm')
 22     cd /Users/Alex/Library/Developer/Xcode/DerivedData/Pent-aedkuecfpwytirevchqirntnielu/SourcePackages/checkouts/realm-cocoa
 23     export LANG\=en_US.US-ASCII

 /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/clang -x c++ -target arm64-apple-ios11.0 -fmessage-length\=0 -fdiagnostics-show-note-include-stac    k -fmacro-backtrace-limit\=0 -std\=c++1z -fmodules -gmodules -fmodules-prune-interval\=86400 -fmodules-prune-after\=345600 -fbuild-session-file\=/var/folders/by/cr9r24wn2fn47wpyjnrq73qc000    0gn/C/org.llvm.clang/ModuleCache.noindex/Session.modulevalidation -fmodules-validate-once-per-build-session -Wnon-modular-include-in-framework-module -Werror\=non-modular-include-in-framew    ork-module -fmodule-name\=Realm -Wno-trigraphs -fpascal-strings -Os -Wno-missing-field-initializers -Wno-missing-prototypes -Wno-return-type -Wno-non-virtual-dtor -Wno-overloaded-virtual -    Wno-exit-time-destructors -Wno-missing-braces -Wparentheses -Wswitch -Wno-unused-function -Wno-unused-label -Wno-unused-parameter -Wno-unused-variable -Wunused-value -Wno-empty-body -Wno-u    ninitialized -Wno-unknown-pragmas -Wno-shadow -Wno-four-char-constants -Wno-conversion -Wno-constant-conversion -Wno-int-conversion -Wno-bool-conversion -Wno-enum-conversion -Wno-float-con    version -Wno-non-literal-null-conversion -Wno-objc-literal-conversion -Wshorten-64-to-32 -Wno-newline-eof -Wno-c++11-extensions -DSWIFT_PACKAGE -DREALM_SPM\=1 -DREALM_COCOA_VERSION\=@\"5.5    .0\" -DREALM_VERSION\=\"6.1.4\" -DREALM_NO_CONFIG -DREALM_INSTALL_LIBEXECDIR\= -DREALM_ENABLE_ASSERTIONS\=1 -DREALM_ENABLE_ENCRYPTION\=1 -DREALM_VERSION_MAJOR\=6 -DREALM_VERSION_MINOR\=1 -    DREALM_VERSION_PATCH\=4 -DREALM_VERSION_EXTRA\=\"\" -DREALM_VERSION_STRING\=\"6.1.4\" -isysroot /Applications/Xcode.app/Contents/Developer/Platforms/iPhoneOS.platform/Developer/SDKs/iPhone    OS14.1.sdk -fstrict-aliasing -Wdeprecated-declarations -Winvalid-offsetof -g -fvisibility-inlines-hidden -Wno-sign-conversion -Wno-infinite-recursion -Wno-move -Wno-comma -Wno-block-captur    e-autoreleasing -Wno-strict-prototypes -Wno-range-loop-analysis -Wno-semicolon-before-method-body -fembed-bitcode-marker -I/Users/Alex/Library/Developer/Xcode/DerivedData/Pent-aedkuecfpwyt    irevchqirntnielu/SourcePackages/checkouts/realm-core/src -I/Users/Alex/Library/Developer/Xcode/DerivedData/Pent-aedkuecfpwytirevchqirntnielu/SourcePackages/checkouts/realm-cocoa/build/Rele    ase-iphoneos/include -I/Users/Alex/Library/Developer/Xcode/DerivedData/Pent-aedkuecfpwytirevchqirntnielu/SourcePackages/checkouts/realm-cocoa/include -I/Users/Alex/Library/Developer/Xcode/    DerivedData/Pent-aedkuecfpwytirevchqirntnielu/SourcePackages/checkouts/realm-cocoa -I/Users/Alex/Library/Developer/Xcode/DerivedData/Pent-aedkuecfpwytirevchqirntnielu/SourcePackages/checko    uts/realm-cocoa/include -I/Users/Alex/Library/Developer/Xcode/DerivedData/Pent-aedkuecfpwytirevchqirntnielu/SourcePackages/checkouts/realm-cocoa/Realm/ObjectStore/src -I/Users/Alex/Library    /Developer/Xcode/DerivedData/Pent-aedkuecfpwytirevchqirntnielu/SourcePackages/checkouts/realm-cocoa/build/Realm.build/Release-iphoneos/Realm.build/DerivedSources-normal/arm64 -I/Users/Alex    /Library/Developer/Xcode/DerivedData/Pent-aedkuecfpwytirevchqirntnielu/SourcePackages/checkouts/realm-cocoa/build/Realm.build/Release-iphoneos/Realm.build/DerivedSources/arm64 -I/Users/Ale    x/Library/Developer/Xcode/DerivedData/Pent-aedkuecfpwytirevchqirntnielu/SourcePackages/checkouts/realm-cocoa/build/Realm.build/Release-iphoneos/Realm.build/DerivedSources -F/Users/Alex/Lib    rary/Developer/Xcode/DerivedData/Pent-aedkuecfpwytirevchqirntnielu/SourcePackages/checkouts/realm-cocoa/build/Release-iphoneos -F/Applications/Xcode.app/Contents/Developer/Platforms/iPhone    OS.platform/Developer/Library/Frameworks -iframework /Applications/Xcode.app/Contents/Developer/Platforms/iPhoneOS.platform/Developer/SDKs/iPhoneOS14.1.sdk/Developer/Library/Frameworks -fm    odule-map-file\=/Users/Alex/Library/Developer/Xcode/DerivedData/Pent-aedkuecfpwytirevchqirntnielu/SourcePackages/checkouts/realm-cocoa/build/GeneratedModuleMaps-iphoneos/RealmCore.modulema    p -DXcode -MMD -MT dependencies -MF /Users/Alex/Library/Developer/Xcode/DerivedData/Pent-aedkuecfpwytirevchqirntnielu/SourcePackages/checkouts/realm-cocoa/build/Realm.build/Release-iphoneo    s/Realm.build/Objects-normal/arm64/collection_notifications.d --serialize-diagnostics /Users/Alex/Library/Developer/Xcode/DerivedData/Pent-aedkuecfpwytirevchqirntnielu/SourcePackages/check    outs/realm-cocoa/build/Realm.build/Release-iphoneos/Realm.build/Objects-normal/arm64/collection_notifications.dia -c /Users/Alex/Library/Developer/Xcode/DerivedData/Pent-aedkuecfpwytirevch    qirntnielu/SourcePackages/checkouts/realm-cocoa/Realm/ObjectStore/src/collection_notifications.cpp -o /Users/Alex/Library/Developer/Xcode/DerivedData/Pent-aedkuecfpwytirevchqirntnielu/Sour    cePackages/checkouts/realm-cocoa/build/Realm.build/Release-iphoneos/Realm.build/Objects-normal/arm64/collection_notifications.o
**fatal error: module map file '/Users/Alex/Library/Developer/Xcode/DerivedData/Pent-aedkuecfpwytirevchqirntnielu/SourcePackages/checkouts/realm-cocoa/build/GeneratedModuleMaps-iphoneos/Real    mCore.modulemap' not found**

Each subsequent error is saying it can't find RealmCore.modulemap in the DerivedData folder.

===========
The following build commands failed:
 75     CompileC /Users/Alex/Library/Developer/Xcode/DerivedData/Pent-aedkuecfpwytirevchqirntnielu/SourcePackages/checkouts/realm-cocoa/build/Realm.build/Release-iphoneos/Realm.build/Objects-n    ormal/arm64/binding_callback_thread_observer.o /Users/Alex/Library/Developer/Xcode/DerivedData/Pent-aedkuecfpwytirevchqirntnielu/SourcePackages/checkouts/realm-cocoa/Realm/ObjectStore/src/    binding_callback_thread_observer.cpp normal arm64 c++ com.apple.compilers.llvm.clang.1_0.compiler
 76     CompileC /Users/Alex/Library/Developer/Xcode/DerivedData/Pent-aedkuecfpwytirevchqirntnielu/SourcePackages/checkouts/realm-cocoa/build/Realm.build/Release-iphoneos/Realm.build/Objects-n    ormal/arm64/keychain_helper.o /Users/Alex/Library/Developer/Xcode/DerivedData/Pent-aedkuecfpwytirevchqirntnielu/SourcePackages/checkouts/realm-cocoa/Realm/ObjectStore/src/impl/apple/keycha    in_helper.cpp normal arm64 c++ com.apple.compilers.llvm.clang.1_0.compiler
 77 (2 failures)

I am using the latest gem: jazzy version: 0.13.5
It seems that since I'm using (remote) packages (and a local package) I am unable to use jazzy.

I don't know how to debug compiler flag issues such as this (the original answer suggested doing this) -- but I'm happy to do so. Can I be pointed in the right direction?

Is there a newer (beta) version that I can try? Or a will a new version of jazzy be available any time soon?

Has anyone else seen this behavior and found a workaround or solution?

**Comment by johnfairh** [comment]
To clarify: it's (probably) not about compiler flags, it's about the build tool flags.

You need to figure out how to build your module from the command line, either with SPM or `xcodebuild`.  Jazzy tries to guess this from various things but it's not working here.

The issues don't have the jazzy command, the project, or any info about its structure so it's tough to give concrete suggestions.  From the error message it could be that you're trying to build a target without its dependencies.  Xcode normally handles this with schemes -- if you're using SPM and aren't sure what this means then `xcodebuild -list` could help.

**Comment by Mozahler** [comment]
Thanks for the quick response, John! I specify the scheme.  (Pent in this case) This is the command line I send to jazzy
+ jazzy '--exclude=build/*' --min-acl private --theme canna --xcodebuild-arguments -target,Pent,-sdk,iphoneos

**Comment by johnfairh** [comment]
Looking at the xcodebuild args above, they don't seem to actually include the scheme name, just the target name?  If your scheme is also called `Pent` then maybe try:
```shell
xcodebuild -scheme Pent -target Pent -sdk iphoneos
```
When you get the `xcodebuild` command working you can pass those same args to jazzy.

**Comment by Mozahler** [comment]
Thanks for clarifying. I'll learn how to compile this project from the command-line as my next step. 👍

**Comment by Mozahler** [comment]
Got it!
The piece of advice that made all the difference was to get it to work without using jazzy first.
It was a little tricky with commas vs. spaces, but once I got the command working on its own it gave me the boost I needed to follow through.

For those who have similar issues, this is what worked:
command line:

xcodebuild -scheme Pent -target Pent -sdk iphoneos

and with jazzy (where 'Pent' is the scheme and target name):

jazzy --exclude=build/* --min-acl private --theme canna --xcodebuild-arguments -scheme,Pent,-sdk,iphoneos

Thanks again, John! ...closing.
